master/internal/api_experiment_intg_test.go

```diff
@@ -0,0 +1,374 @@
+//go:build integration
+// +build integration
+
+package internal
+
+import (
+	"context"
+	"sort"
+	"testing"
+	"time"
+	"unsafe"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/ptrs"
+	"github.com/determined-ai/determined/master/pkg/schemas"
+	"github.com/determined-ai/determined/master/pkg/schemas/expconf"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/experimentv1"
+	"github.com/determined-ai/determined/proto/pkg/userv1"
+)
+
+var minExpConfig = expconf.ExperimentConfig{
+	RawResources: &expconf.ResourcesConfig{
+		RawResourcePool: ptrs.Ptr("kubernetes"),
+	},
+	RawEntrypoint: &expconf.EntrypointV0{"test"},
+	RawCheckpointStorage: &expconf.CheckpointStorageConfig{
+		RawSharedFSConfig: &expconf.SharedFSConfig{
+			RawHostPath: ptrs.Ptr("/"),
+		},
+	},
+	RawHyperparameters: expconf.Hyperparameters{},
+	RawReproducibility: &expconf.ReproducibilityConfig{ptrs.Ptr(uint32(42))},
+	RawSearcher: &expconf.SearcherConfig{
+		RawMetric: ptrs.Ptr("loss"),
+		RawSingleConfig: &expconf.SingleConfig{
+			&expconf.Length{Units: 10, Unit: "batches"},
+		},
+	},
+}
+
+func TestGetExperiments(t *testing.T) {
+	// Setup.
+	api, _, ctx := SetupAPITest(t)
+	_, err := db.Bun().NewTruncateTable().Table("experiments").Cascade().Exec(ctx)
```

Comment:
This will delete all of your experiments when running an integration test which may or may not be okay?

Not sure how to get around this besides like always filtering on some random field

Tried renaming ```experiments``` then copying it but that ran into a bunch of other issues like the IDs not generating then not being able to create trials to test with